src/main/java/TaskList.java

```diff
@@ -0,0 +1,158 @@
+import java.util.Scanner;
+public class TaskList {
+        private final Task[] taskList;
+        private static final int MAX_TASKS = 100;
+        //protected boolean[] isMarked;
+        private int taskCount;
+
+
+        public TaskList() {
+            this.taskList = new Task[MAX_TASKS];
+            this.taskCount = 0;
+        }
+
+    private void processTaskCommand(String command) {
+        String[] commandParts = command.split(" ", 2);
+        String taskType = commandParts[0];
+
+        switch (taskType) {
+        case "todo":
+            addTask(new Task(commandParts[1]));
+            break;
+        case "deadline":
+            String[] deadlineParts = commandParts[1].split("/by" , 2);
+            if (deadlineParts.length != 2) {
+                System.out.println("Invalid deadline format! Use: deadline <<description>> /by <<deadline>>.");
+            } else {
+                addTask(new Deadline(deadlineParts[0], deadlineParts[1]));
+            }
+            break;
+        case "event":
+            String[] eventparts = commandParts[1].split("/from" , 2);
+            if (eventparts.length != 2) {
+                System.out.println("Invalid deadline format! Use: deadline <<description>> /by <<deadline>>.");
+            } else {
+                String[] endDateParts = eventparts[1].split("/to" , 2);
+                if (endDateParts.length != 2) {
```

Comment:
You could consider refactoring this nested conditional to prevent deep nesting and adhere to this code quality [guideline](https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-deep-nesting)